src/sync.rs

```diff
@@ -214,16 +214,46 @@ impl<T> GILOnceCell<Py<T>> {
     }
 }
 
-impl GILOnceCell<Py<PyType>> {
+impl<T> GILOnceCell<Py<T>>
+where
+    T: PyTypeCheck,
+{
     /// Get a reference to the contained Python type, initializing it if needed.
     ///
     /// This is a shorthand method for `get_or_init` which imports the type from Python on init.
-    pub(crate) fn get_or_try_init_type_ref<'py>(
+    ///
+    /// # Example: Using `GILOnceCell` to avoid the overhead of importing a class multiple times
+    ///
```

Comment:
It is better to just say what the example does, this makes for nicer URLs as well.

```suggestion
    /// # Example: Using `GILOnceCell` to store a class in a static variable.
    ///
    /// `GILOnceCell` can be used to avoid importing a class multiple times
```

(this also explicitly mentions "static", which is important. We don't want people to have local (or worse, `const`) GILOnceCells).

Reply:
Done: https://github.com/PyO3/pyo3/pull/4542/commits/5d95fbad3bac21a900936eb65cc20fb076751cf8